apps/mobile/src/components/ProfileView/ProfileViewSharedInfo/ProfileViewSharedCommunities.tsx

```diff
@@ -0,0 +1,175 @@
+import { removeNullValues } from '~/shared/relay/removeNullValues';
+
+import {
+  ContractAddress,
+  ProfileViewSharedCommunitiesSheet,
+} from './ProfileViewSharedCommunitiesSheet';
+
+type Props = {
+  userRef: ProfileViewSharedCommunitiesFragment$key;
+};
+
+export const SHARED_COMMUNITIES_PER_PAGE = 20;
+
+export default function ProfileViewSharedCommunities({ userRef }: Props) {
+  const user = useFragment(
+    graphql`
+      fragment ProfileViewSharedCommunitiesFragment on GalleryUser {
+        __typename
+        sharedCommunities(first: $sharedCommunitiesFirst, after: $sharedCommunitiesAfter)
+          @connection(key: "UserSharedInfoFragment_sharedCommunities") {
+          edges {
+            node {
+              __typename
+              ... on Community {
+                __typename
+                name
+                chain
+                contractAddress {
+                  address
+                  chain
+                }
+              }
+            }
+          }
+          pageInfo {
+            total
+          }
+        }
+        ...ProfileViewSharedCommunitiesSheetFragment
+      }
+    `,
+    userRef
+  );
+
+  const sharedCommunities = useMemo(() => {
+    const list = user.sharedCommunities?.edges?.map((edge) => edge?.node) ?? [];
+    return removeNullValues(list);
+  }, [user.sharedCommunities?.edges]);
+
+  const totalSharedCommunities = user.sharedCommunities?.pageInfo?.total ?? 0;
+
+  // Determine how many users to display by username
+  const communitiesToDisplay = useMemo(() => {
+    // In most cases we display a max of 2 communities. ie "community1, community2 and 3 others"
+    // But if there are exactly 3 shared communities, we display all 3. ie "community1, community2 and community3"
+    const maxNamesToDisplay = totalSharedCommunities === 3 ? 3 : 2;
+    return sharedCommunities.slice(0, maxNamesToDisplay);
+  }, [sharedCommunities, totalSharedCommunities]);
+
+  const bottomSheetRef = useRef<GalleryBottomSheetModalType | null>(null);
+  const handleSeeAllPress = useCallback(() => {
+    bottomSheetRef.current?.present();
+  }, []);
+
+  const navigation = useNavigation<MainTabStackNavigatorProp>();
+
+  const handleCommunityPress = useCallback(
+    (contractAddress: ContractAddress) => {
+      const { address, chain } = contractAddress ?? {};
+
+      if (!address || !chain) return;
+      navigation.push('Community', {
+        contractAddress: address,
+        chain,
+      });
+    },
+    [navigation]
+  );
+
+  const content = useMemo(() => {
+    // Display up to 3 communities
+    const result = communitiesToDisplay.map((community) => {
+      if (!!community.contractAddress && community.chain) {
+        return (
+          <InteractiveLink
+            onPress={() =>
+              community.contractAddress && handleCommunityPress(community.contractAddress)
+            }
+            key={community.name}
+          >
```

Comment:
hmm would be cool to track this. technically we do in `InteractiveLink` but because there's no `href` provided we won't know exactly what was clicked

after this gets merged i can make an update where you can pass in tracking props to `InteractiveLink`

Reply:
good idea , in that case is it cool to leave as is for this PR?

Reply:
yessir!